profiling-ffi/src/exporter.rs

```diff
@@ -12,19 +12,18 @@ use ddcommon::tag::Tag;
 use ddcommon_ffi::slice::{AsBytes, ByteSlice, CharSlice, Slice};
 use ddcommon_ffi::Error;
 use std::borrow::Cow;
-use std::ptr::NonNull;
+use std::ptr::{drop_in_place, NonNull};
 use std::str::FromStr;
 
 #[repr(C)]
 pub enum ExporterNewResult {
-    Ok(*mut ProfileExporter),
+    Ok(NonNull<ProfileExporter>),
     Err(Error),
 }
```

Comment:
This changes the type exported in the FFI from `struct ddog_prof_Exporter_Request *ok` to `struct ddog_prof_NonNull_Request ok`.

I'm not sure I quite follow this change -- doesn't this `NonNull_Request` become null after being used/consumed?

(Also we probably need to update `cbindgen.toml` to call this... something. Perhaps this should still be named a `ddog_prof_Exporter_Request` as far as C clients care?)

Reply:
> This changes the type exported in the FFI from struct ddog_prof_Exporter_Request *ok to struct ddog_prof_NonNull_Request ok.

This is not happening for me -- I double checked this. I shall triple-check!

Reply:
As far as the `NonNull<T>` being accurate... yes, I think it is:

1. At the point these `*Result` types are created in Rust, they are not null. When using the Rust API to test these things, we get a nicer type to work with.
2. These `*Result` types are never _consumed_ by a Rust API except in tests, so there isn't an opportunity for `extern "C"` users to screw this up.

Does that sound right to you?

Reply:
I've just tried with the latest cbindgen (I was using the one from the ubuntu repos) and it looks good with that one :+1: